### DISCUSSION TITLE
Support for persisted queries?

### DISCUSSION THREAD

**Discussion by EmilioLizama** [ExpediaGroup/graphql-kotlin#1988] [Ideas]
Hello, is there any support for the use of Persisted Queries? I don't mean Automatic Persisted Queries (APQ) as I noticed it is already supported. This is mostly because I want to have a little bit more security, and the clients can only do a handful of queries and can't create any more than the ones that are permitted.

**Comment by samuelAndalon** [comment]
please have a look at https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/execution/preparsed/persisted/PersistedQuerySupport.java, you can write your own implementation of that Abstract class and provide your own preparsed document provider with your prefilled cache.
